pkg/cmd/experimental/syncgroups/groupsyncer.go

```diff
@@ -0,0 +1,141 @@
+package syncgroups
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-ldap/ldap"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+	ouserapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// GroupSyncer runs a Sync job on Groups
+type GroupSyncer interface {
+	Sync() (errors []error)
+}
+
+// LDAPGroupSyncer sync Groups with records on an external LDAP server
+type LDAPGroupSyncer struct {
```

Comment:
Unit tests for this type.  Stub in test structs for the values and make sure the flows are correct and errors are handled properly.